static/js/stream_popover.js

```diff
@@ -414,46 +426,78 @@ function build_move_topic_to_stream_popover(e, current_stream_id, topic_name) {
         // Unlike most topic comparisons in Zulip, we intentionally do
         // a case-sensitive comparison, since adjusting the
         // capitalization of a topic is a valid operation.
+        // new_topic_name can be undefined when the new topic input is
+        // disabled in case when user does not have permission to edit
+        // topic and thus submit button is disabled if stream is also
+        // not changed.
         $("#move_topic_modal .dialog_submit_button")[0].disabled =
             Number.parseInt(current_stream_id, 10) === Number.parseInt(select_stream_id, 10) &&
-            new_topic_name.trim() === old_topic_name.trim();
+            (new_topic_name === undefined || new_topic_name.trim() === old_topic_name.trim());
     }
 
     function move_topic() {
         const params = get_params_from_form();
 
         const {old_topic_name} = params;
-        const select_stream_id = stream_widget.value();
+        let select_stream_id = stream_widget.value();
 
         let {
             current_stream_id,
             new_topic_name,
             send_notification_to_new_thread,
             send_notification_to_old_thread,
         } = params;
-        new_topic_name = new_topic_name.trim();
         send_notification_to_new_thread = send_notification_to_new_thread === "on";
         send_notification_to_old_thread = send_notification_to_old_thread === "on";
         current_stream_id = Number.parseInt(current_stream_id, 10);
 
+        if (new_topic_name !== undefined) {
+            // new_topic_name can be undefined when the new topic input is disabled when
+            // user does not have permission to edit topic.
+            new_topic_name = new_topic_name.trim();
+        }
+        if (old_topic_name.trim() === new_topic_name) {
+            // We use `undefined` to tell the server that
+            // there has been no change in the topic name.
+            new_topic_name = undefined;
+        }
+
+        let propagate_mode = "change_all";
+        if (message !== undefined) {
+            if (select_stream_id === current_stream_id) {
+                // We use `undefined` to tell the server that
+                // there has been no change in stream. This is
+                // important for cases when changing stream is
+                // not allowed.
+                select_stream_id = undefined;
+            }
```

Comment:
We can move this outside the `message !== undefined` if we can be sure that the [below condition is not necessary](https://github.com/zulip/zulip/pull/23139/files#r989309707).